packages/legacy/core/App/screens/PasteUrl.tsx

```diff
@@ -0,0 +1,139 @@
+import { useAgent } from '@credo-ts/react-hooks'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View, TextInput, ScrollView, TouchableOpacity, Modal } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+
+import Button, { ButtonType } from '../components/buttons/Button'
+import InfoBox, { InfoBoxType } from '../components/misc/InfoBox'
+import { TOKENS, useContainer } from '../container-api'
+import { useConfiguration } from '../contexts/configuration'
+import { useTheme } from '../contexts/theme'
+import { ConnectStackParams } from '../types/navigators'
+import { connectFromScanOrDeepLink } from '../utils/helpers'
+import { testIdWithKey } from '../utils/testable'
+
+export type PasteProps = StackScreenProps<ConnectStackParams>
+
+const PasteUrl: React.FC<PasteProps> = ({ navigation }) => {
+  const { ColorPallet, TextTheme } = useTheme()
+  const [pastedContent, setPastedContent] = useState<string>('')
+  const [errorMessage, setErrorMessage] = useState<{ title: string; message: string } | undefined>()
+  const { t } = useTranslation()
+  const { agent } = useAgent()
+  const container = useContainer()
+  const logger = container.resolve(TOKENS.UTIL_LOGGER)
+  const { enableImplicitInvitations, enableReuseConnections } = useConfiguration()
+  const styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      backgroundColor: ColorPallet.brand.primaryBackground,
+    },
+    content: {
+      margin: 20,
+    },
+    description: {
+      ...TextTheme.normal,
+      marginBottom: 20,
+    },
+    textBox: {
+      ...TextTheme.normal,
+      textAlignVertical: 'top',
+      borderColor: ColorPallet.grayscale.darkGrey,
+      borderWidth: 1,
+      borderRadius: 5,
+      backgroundColor: ColorPallet.grayscale.lightGrey,
+    },
+    buttonContainer: {
+      margin: 20,
+    },
+  })
+
+  const processPastedContent = async () => {
+    try {
+      await connectFromScanOrDeepLink(
+        pastedContent,
+        agent,
+        logger,
+        navigation?.getParent(),
+        false, // isDeepLink
+        enableImplicitInvitations,
+        enableReuseConnections
+      )
+    } catch (err: unknown) {
+      setErrorMessage({ title: t('PasteUrl.ErrorInvalidUrl'), message: t('PasteUrl.ErrorInvalidUrlDescription') })
```

Comment:
Should we include `err?.message` in here anywhere to provide more context as to why it's failing?

Reply:
a little bit difficult, there's multiple different errors that can result from an invalid url. It could be already scanned, it could contain incorrect parameters, invalid url scheme, invalid json, invalid invitation object, etc. All of these create a different error message. it's difficult to accurately detect what is failing due to a malformed url, and what is failing due to some other unknown error and if we should print the error.